keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -148,3 +148,173 @@ def get_config(self):
             }
         )
         return config
+
+    def _augment(self, inputs):
+        image = inputs.get("images", None)
+        label = inputs.get("labels", None)
+        bounding_box = inputs.get("bounding_boxes", None)
+        transformation = self.get_random_transformation(
+            image=image, label=label, bounding_box=bounding_box
+        )  # pylint: disable=assignment-from-none
+        image = self.augment_image(image, transformation=transformation)
+        result = {"images": image}
+        if label is not None:
+            label = self.augment_label(label, transformation=transformation)
+            result["labels"] = label
+        if bounding_box is not None:
+            # changed augment_bounding_box requires image as well
+            # for unnormalizing bbox coordinates
+            bounding_box = self.augment_bounding_box(
+                image, bounding_box, transformation=transformation
+            )
+            result["bounding_boxes"] = bounding_box
+        return result
+
+    def augment_bounding_box(self, image, bounding_boxes, transformation):
+        """args: image : takes a single image H,W,C,
+        bounding_boxes: take bbox coordinates [N,4] -> [y1,x1,y2,x2]
+        transformation: takes tuple for x,y transformation None if no transformation"""
+        x, y = transformation
+        if x is not None:
+            bounding_boxes = self.augment_horizontal(image, bounding_boxes, x)
```

Comment:
2 notes:

- self.augment_horizontal should be a private method (self._augment_horizontal).
- self.augment_horizontal is ambigous.  Please rename to something involving bounding boxes.